test/000_fullchain.test.ts

```diff
@@ -99,7 +105,109 @@ describe('IexecPoco (IT)', function () {
         };
     }
 
-    describe('MatchOrders', function () {
+    it('[1] Sponsorship, beneficiary, callback, BoT, replication', async function () {
+        const volume = 3;
+        // Create deal.
+        const orders = buildOrders({
+            assets: ordersAssets,
+            prices: ordersPrices,
+            requester: requester.address,
+            tag: standardDealTag,
+            beneficiary: beneficiary.address,
+            callback: callbackAddress,
+            volume,
+            trust: 1, // TODO use 5 workers.
+        });
+        const { dealId, dealPrice, schedulerStakePerDeal } =
+            await iexecWrapper.signAndSponsorMatchOrders(...orders.toArray());
+        const taskPrice = appPrice + datasetPrice + workerpoolPrice;
+        const schedulerStakePerTask = schedulerStakePerDeal / volume;
+        const workerRewardPerTask = await iexecWrapper.computeWorkerRewardPerTask(
+            dealId,
+            PocoMode.CLASSIC,
+        );
+        const schedulerRewardPerTask = workerpoolPrice - workerRewardPerTask;
+        // Check initial balances.
+        await checkBalancesAndFrozens({
+            proxyBalance: dealPrice + schedulerStakePerDeal,
+            accounts: [
+                { signer: sponsor, balance: 0, frozen: dealPrice },
+                { signer: requester, balance: 0, frozen: 0 },
+                { signer: scheduler, balance: 0, frozen: schedulerStakePerDeal },
+                { signer: appProvider, balance: 0, frozen: 0 },
+                { signer: datasetProvider, balance: 0, frozen: 0 },
+                { signer: worker1, balance: 0, frozen: 0 },
+            ],
+        });
+        // Finalize each task and check balance changes.
+        for (let taskIndex = 0; taskIndex < volume; taskIndex++) {
+            const taskId = await iexecWrapper.initializeTask(dealId, taskIndex);
+            const { workerStakePerTask } = await iexecWrapper.contributeToTask(
+                dealId,
+                taskIndex,
+                callbackResultDigest,
+                worker1,
+            );
+            await iexecPoco
+                .connect(worker1)
+                .reveal(taskId, callbackResultDigest)
+                .then((tx) => tx.wait());
+            await iexecPoco
+                .connect(scheduler)
+                .finalize(taskId, results, resultsCallback)
+                .then((tx) => tx.wait());
+            expect((await iexecPoco.viewTask(taskId)).status).to.equal(TaskStatusEnum.COMPLETED);
+            // Multiply amount by the number of finalized tasks to correctly compute
+            // stake and reward amounts.
+            const timesTasks = (amount: number) => amount * (taskIndex + 1);
```

Comment:
What do you think of a different name ? `scaledByTaskCount` instead of `timesTasks` (?)

Reply:
```suggestion
            const completedTasks = taskIndex + 1;
```
then `balance: value * completedTasks`

Reply:
Implemented @jeremyjams suggestion.